core/src/logging/LogMasker.ts

```
@@ -18,99 +18,86 @@
-  }
-
-  protected maskSubstring (): string {
-    return LoggingMessage.OMITTED
-  }
+export function maskResponse (config: AxiosResponse<any, any>): AxiosResponse<any, any> {
+  const clone: AxiosResponse<any, any> = _.cloneDeep(config)
+  clone.headers = maskResponseHeaders(clone.headers)
+  clone.data = maskBodyFields(clone.data)
+  clone.config.auth = maskAuthCredentials(clone.config.auth)
+  clone.config.data = LoggingMessage.OMITTED
+  delete clone.request
+  return clone
 }
 
-class AuthTokenMask extends Mask {
-  private static readonly _instance: AuthTokenMask = new AuthTokenMask()
-
-  protected readonly regex: RegExp = new RegExp(LogMaskingRegex.AUTHORIZATION_REGEX)
-
-  static get instance (): AuthTokenMask {
-    return this._instance
-  }
-
-  private constructor () {
-    super()
+function stringifyData (data: any): any {
+  if (data === undefined) {
+    return data
   }
+  return JSON.stringify(Serializer.serialize(data))
 }
 
-class AuthUsernameMask extends Mask {
-  private static readonly _instance: AuthUsernameMask = new AuthUsernameMask()
-
-  protected readonly regex: RegExp = new RegExp(LogMaskingRegex.AUTH_USERNAME_REGEX)
-
-  static get instance (): AuthUsernameMask {
-    return this._instance
+function maskRequestConfigHeaders (headers: AxiosRequestHeaders): AxiosRequestHeaders {
+  const clonedHeaders: AxiosRequestHeaders = new AxiosHeaders()
+  for (const header in headers) {
+    clonedHeaders[header] = LOG_MASKING_HEADERS.includes(header.toLowerCase())
+      ? LoggingMessage.OMITTED
+      : headers[header]
   }
+  return clonedHeaders
 }
 
-class AuthPasswordMask extends Mask {
-  private static readonly _instance: AuthPasswordMask = new AuthPasswordMask()
-
-  protected readonly regex: RegExp = new RegExp(LogMaskingRegex.AUTH_PASSWORD_REGEX)
-
-  static get instance (): AuthPasswordMask {
-    return this._instance
+function maskAuthCredentials (auth: AxiosBasicCredentials | undefined): AxiosBasicCredentials | undefined {
```

Do we only support basic credentials?

---

Yes

---

Shouldn't we, at least, add some checks before masking it? If we have basic credentials mask it by this method, otherwise throw an exception.

---

Only auth requests should contain basic credentials, other future requests will contain an auth token in the Authorization header. So requests are bound to both scenarios. I'm not sure if I got you correctly, why do we need to throw an exception?